### PR TITLE
[CARBONDATA-2260] CarbonThriftServer should support store carbon table on S3

### PR DESCRIPTION
**Pull request by xubo245**
CarbonThriftServer should support store carbon table on S3
 - Add config for AK,SK,EndPoint when start carbonsession

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [x] Any backward compatibility impacted?
 
 - [x] Document update required?

 - [x] Testing done
        Step 1: 
          start carbonThriftServer by  org.apache.carbondata.examples.CarbonThriftServerExample
          User should use themselves storePath, AK, SK and endPoint
      Step 2: 
          start beeline like:  beeline -u jdbc:hive2://IP:10000
         IP and port should provide
       Now it support running with local, HDFS and S3 mode
Test example:

```
root@ecs-909c:~# beeline -u jdbc:hive2://192.168.0.206:10000
Connecting to jdbc:hive2://192.168.0.206:10000
2018-03-17 17:44:29 INFO  Utils:310 - Supplied authorities: 192.168.0.206:10000
2018-03-17 17:44:29 INFO  Utils:397 - Resolved authority: 192.168.0.206:10000
2018-03-17 17:44:29 INFO  HiveConnection:203 - Will try to open client transport with JDBC Uri: jdbc:hive2://192.168.0.206:10000
Connected to: Spark SQL (version 2.2.1)
Driver: Hive JDBC (version 1.2.1.spark2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
Beeline version 1.2.1.spark2 by Apache Hive
0: jdbc:hive2://192.168.0.206:10000>  Drop table if exists local_table2;
+---------+--+
| Result  |
+---------+--+
+---------+--+
No rows selected (21.205 seconds)
0: jdbc:hive2://192.168.0.206:10000>
0: jdbc:hive2://192.168.0.206:10000> CREATE TABLE if not exists local_table2( shortField SHORT, intField INT, bigintField LONG, doubleField DOUBLE, stringField STRING, timestampField TIMESTAMP, decimalField DECIMAL(18,2), dateField DATE, charField CHAR(5), floatField FLOAT ) STORED BY 'carbondata' LOCATION 's3a://carbonstore/local_table2'  TBLPROPERTIES('SORT_COLUMNS'='');
+---------+--+
| Result  |
+---------+--+
+---------+--+
No rows selected (1.296 seconds)
0: jdbc:hive2://192.168.0.206:10000>
0: jdbc:hive2://192.168.0.206:10000>
0: jdbc:hive2://192.168.0.206:10000> LOAD DATA LOCAL INPATH '/huawei/xubo/git/carbondata2/examples/spark2/src/main/resources/data1.csv' INTO TABLE local_table2 OPTIONS('HEADER'='true');
+---------+--+
| Result  |
+---------+--+
+---------+--+
No rows selected (3.339 seconds)
0: jdbc:hive2://192.168.0.206:10000>
0: jdbc:hive2://192.168.0.206:10000>  select * from local_table2;
+-------------+-----------+--------------+--------------+--------------+------------------------+---------------+-------------+------------+-------------+--+
| shortfield  | intfield  | bigintfield  | doublefield  | stringfield  |     timestampfield     | decimalfield  |  datefield  | charfield  | floatfield  |
+-------------+-----------+--------------+--------------+--------------+------------------------+---------------+-------------+------------+-------------+--+
| 1           | 10        | 1100         | 48.4         | spark        | 2015-04-23 12:01:01.0  | 1.23          | 2015-04-23  | aaa        | 2.5         |
| 5           | 17        | 1140         | 43.4         | spark        | 2015-07-27 12:01:02.0  | 3.45          | 2015-07-27  | bbb        | 2.5         |
| 1           | 11        | 1100         | 44.4         | flink        | 2015-05-23 12:01:03.0  | 23.23         | 2015-05-23  | ccc        | 2.5         |
| 1           | 10        | 1150         | 43.4         | spark        | 2015-07-24 12:01:04.0  | 254.12        | 2015-07-24  | ddd        | 2.5         |
| 1           | 10        | 1100         | 47.4         | spark        | 2015-07-23 12:01:05.0  | 876.14        | 2015-07-23  | eeee       | 3.5         |
| 3           | 14        | 1160         | 43.4         | hive         | 2015-07-26 12:01:06.0  | 3454.32       | 2015-07-26  | ff         | 2.5         |
| 2           | 10        | 1100         | 43.4         | impala       | 2015-07-23 12:01:07.0  | 456.98        | 2015-07-23  | ggg        | 2.5         |
| 1           | 10        | 1100         | 43.4         | spark        | 2015-05-23 12:01:08.0  | 32.53         | 2015-05-23  | hhh        | 2.5         |
| 4           | 16        | 1130         | 42.4         | impala       | 2015-07-23 12:01:09.0  | 67.23         | 2015-07-23  | iii        | 2.5         |
| 1           | 10        | 1100         | 43.4         | spark        | 2015-07-23 12:01:10.0  | 832.23        | 2015-07-23  | jjj        | 2.5         |
+-------------+-----------+--------------+--------------+--------------+------------------------+---------------+-------------+------------+-------------+--+
10 rows selected (0.959 seconds)
0: jdbc:hive2://192.168.0.206:10000>


```
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4396/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3164/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4397/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3920/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4411/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3177/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3932/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4414/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3180/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3935/

**Comment by xubo245** [comment]
@jackylk Please review it!

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala [hunk 1]
```diff
@@ -34,10 +36,23 @@ object CarbonThriftServer {
     import org.apache.spark.sql.CarbonSession._
 
     val sparkConf = new SparkConf(loadDefaults = true)
+
+    val logger: Logger = LoggerFactory.getLogger(this.getClass)
+    if (args.length != 1 && args.length != 4) {
+      logger.error("parameters: storePath [access-key] [secret-key] [s3-endpoint]")
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: After this change, how user can use this class for HDFS cluster?

Comment (xubo245) [reply integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: Fix it, now it support local, HDFS and S3 mode

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4508/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3280/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4015/

**Comment by xubo245** [comment]
retest this please

**Comment by xubo245** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3286/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4514/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4019/

**Comment by xubo245** [comment]
retest sdv please

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3300/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4528/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4028/

**Comment by xubo245** [comment]
@jackylk Please review it

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3412/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4636/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4146/

**Comment by xubo245** [comment]
@jackylk please review it.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3724/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4943/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5046/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5047/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3828/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3831/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3870/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5096/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5106/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5110/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3893/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5114/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3895/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5116/

**Comment by xubo245** [comment]
@jackylk  Please review it

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5164/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3954/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4040/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5249/

**Comment by xubo245** [comment]
@jackylk Please review it.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4171/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4486/

**Comment by xubo245** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4499/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4504/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5361/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4211/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5379/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4216/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5384/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**